release/ray_release/test.py

```diff
@@ -464,22 +466,38 @@ def get_test_results(
             key=lambda file: int(file["LastModified"].timestamp()),
             reverse=True,
         )[:limit]
-        self.test_results = [
-            TestResult.from_dict(
-                json.loads(
-                    s3_client.get_object(
-                        Bucket=bucket,
-                        Key=file["Key"],
-                    )
-                    .get("Body")
-                    .read()
-                    .decode("utf-8")
-                )
+        loop = asyncio.get_event_loop()
+        self.test_results = loop.run_until_complete(
```

Comment:
you would want to async all the way, where a loop is created in main, so that other async functions can share the loop.

Reply:
ideally all functions in our code base should be async yes, this would require a big refactoring of the entire code base (took 2 years for facebook to do it) so i pause the async chain at the `get_test_results` level for now

Reply:
or you mean the event loop needs to be unique per async function?

Reply:
look like python 3.8+ provide another abstraction for doing this, one sec